src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentProjectsFinder.cs

```diff
@@ -300,154 +217,75 @@ private static bool IsInternalsVisibleToAttribute(AttributeData attr)
             HashSet<DependentProject> dependentProjects,
             CancellationToken cancellationToken)
         {
-            var isSubmission = sourceProject != null && sourceProject.IsSubmission;
-            if (isSubmission)
+            if (symbolOrigination.sourceProject?.IsSubmission == true)
                 return;
 
-            var internalsVisibleToMap = CreateInternalsVisibleToMap(sourceAssembly);
-
-            var sourceAssemblySymbolKey = sourceAssembly.GetSymbolKey(cancellationToken);
-
-            // TODO(cyrusn): What about error tolerance situations.  Do we maybe want to search
-            // transitive dependencies as well?  Even if the code wouldn't compile, they may be
-            // things we want to find.
-            foreach (var projectId in solution.ProjectIds)
+            // Set of assembly names that `assembly` has IVT to.  Computed on demand once needed.
+            HashSet<string>? internalsVisibleToSet = null;
+            foreach (var project in solution.Projects)
             {
-                var project = solution.GetRequiredProject(projectId);
-
-                cancellationToken.ThrowIfCancellationRequested();
-
-                if (project.SupportsCompilation && HasReferenceTo(sourceAssembly, sourceProject, project, cancellationToken))
+                if (!project.SupportsCompilation ||
+                    !HasReferenceTo(symbolOrigination, project, cancellationToken))
                 {
-                    var hasInternalsAccess = await HasInternalsAccessAsync(
-                        sourceAssembly, internalsVisibleToMap,
-                        sourceAssemblySymbolKey, project, cancellationToken).ConfigureAwait(false);
-
-                    dependentProjects.Add(new DependentProject(project.Id, hasInternalsAccess));
+                    continue;
                 }
-            }
-        }
 
-        private static async Task<bool> HasInternalsAccessAsync(
-            IAssemblySymbol sourceAssembly, Lazy<HashSet<string>> internalsVisibleToMap,
-            SymbolKey sourceAssemblySymbolKey, Project project, CancellationToken cancellationToken)
-        {
-            if (internalsVisibleToMap.Value.Contains(project.AssemblyName) &&
-                project.SupportsCompilation)
-            {
-                var compilation = await project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
-
-                var targetAssembly = compilation.Assembly;
-                if (sourceAssembly.Language != targetAssembly.Language)
-                {
-                    var resolvedSymbol = sourceAssemblySymbolKey.Resolve(compilation, cancellationToken: cancellationToken).Symbol;
-                    if (resolvedSymbol is IAssemblySymbol sourceAssemblyInTargetCompilation)
-                    {
-                        return targetAssembly.IsSameAssemblyOrHasFriendAccessTo(sourceAssemblyInTargetCompilation);
```

Comment:
If we're no longer checking this, do we still do this somewhere in FAR? I'm imagining a case where:

1. Project B depends on A, which grants IVT to an assembly with the same name but the compiler says no.
2. I find refs on some symbol in A.

We can't call the compiler method to check until we have both compilations of course, but by dropping this are we going to start doing further checks (binding/tree walking) that would have been avoided in this other case?

Reply:
yes.  we will def do extra checks in that case.  i'm weighing off the chance of that happening (and the consequences) against all this work that we're doing now.  i.e. we do 45+ seconds worth of computation to try to just figure out the precise set, insetad of jsut having a slightly larger set and potentially have a few more hits we might have to filter out.

How likely do we think it is that someone has IVT to a same-named project that hte compiler says 'no' to?  wouldn't that be the rarer and less expected case?

Reply:
Oh I'm all for making this the cheap and approximate check. I just wasn't sure if FAR still does call IsSameAssemblyOrHasFriendAccessTo _after_ we have the Compilation but before we actually walk trees and do other expensive stuff. I'd hate to send FAR down a wild goose chase if it was missing the check.

Reply:
I also have no sense in the "real world" what sorts of strangeness is out there.

Reply:
> Oh I'm all for making this the cheap and approximate check. I just wasn't sure if FAR still does call IsSameAssemblyOrHasFriendAccessTo after we have the Compilation but before we actually walk trees and do other expensive stuff. I'd hate to send FAR down a wild goose chase if it was missing the check.

No, it doesn't do that check later.  We do always check that the reference is a true reference.

> I also have no sense in the "real world" what sorts of strangeness is out there.

I think having the uncommon cases pay this price instead of having everyone pay this price is desirable.  

Again, for this to trigger, you would need ot have such a project arrangement where you had a project that said it had IVT to another project with the same name, but didn't intend for it to see internals.  And in that case, all that would happen is that we will end up checking that project, where the majority of the time will be getting the compilation for it.  And that's the cost we're already paying *today*, just that we pay it no matter what.  

i.e. today we go to all these potential matches, and we then make a compilation (the most expensive part of find refs) just to ask "should we really check you?".  This PR says: yes, we should check you, and pushes the getting of the compilation until teh point that we really actually would need to bind and determine if a symbol matches.  

So, in the case where the downstream project *does* reference the name that we're looking for, we pay the same price.  But in the common case that the downstream project doesn't, we are much much cheaper.  

i.e. if i'm searching for a public symbol from MS.CodeAnalysis, then wasting all this computation to determine if we have IVT is irrelevant and wasteful.  *And*, if i'm searching for an internal symbol, then precomputing the compilation for all downstream projects prior to even determining if there's a potential match in them is also wasteful.  

Currently, getting the compilation is the most expensive part of FAR.  Anything that pushes that off until absolutely necessary is seriously beneficial.